pkgs/applications/science/math/caffe/default.nix

```diff
@@ -0,0 +1,70 @@
+{ stdenv
+, openblas
+, boost
+, cudaSupport ? true
+, cudnnSupport ? false
+, cudnn ? null
+, cudatoolkit7
+, fetchzip
+, google-gflags
+, glog
+, hdf5
+, leveldb
+, lmdb
+, opencv
+, protobuf
+, snappy
+}:
+
+
+let optional = stdenv.lib.optional;
+in stdenv.mkDerivation rec {
+  # Use git revision because latest "release" is really old
+  name = "caffe-git-2015-07-02";
+  rev = "77d66dfc";
+
+  src = fetchzip {
```

Comment:
fetchFromGithub does exactly the same, but it's interface is more human readable :)